frontend/js/maze1.js

```diff
@@ -0,0 +1,297 @@
+'use strict';
+import * as polyfills from'./polyfills';
+polyfills.installMatches(); //cross browser polyfill for 'matches' (does not supported by IE)
+polyfills.installClosest(); //cross browser polyfill for 'closest' (does not supported by IE)
+polyfills.installCustomEvent(); //cross browser polyfill for 'custom events' (does not supported by IE)
+
+export default class Maze {
+    constructor(options) {
+        this._el = options.element;
+        this._size = options.size;
+        this._neighborhoodCondition = options.withDiagonalNeighbors ? Maze.NEIGHBORHOOD.ORTHO_DIAGONAL : Maze.NEIGHBORHOOD.ORTHOGONAL;
+
+        this._renderMaze();
+
+        this._map = this._el.querySelector('[data-selector="map"]');
+        this._map.addEventListener('click', Maze._changeCell);
+
+        this._startPoint = undefined;
+        this._endPoint = undefined;
+        this._step = undefined;
```

Comment:
Tell me please, what is better to do: set these fields equal to ‘null’ or just declare them?